### PR TITLE
add zlib as a run-dependency on windows

### PR DESCRIPTION
**Pull request by h-vetinari**
Fixing https://github.com/conda-forge/scipy-feedstock/pull/169/commits/d94e69a9bd87a15d0d708a0bc22afdc2be72dde3 (though not 100% sure I got this right @isuruf)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -225,7 +225,8 @@ outputs:
         - llvm =={{ version }}
         - {{ pin_compatible("libclang-cpp" + minor_aware_ext, exact=True) }}  # [unix]
       run:
-        - {{ pin_compatible("libcxx", max_pin=None) }}      # [osx]
+        - zlib                                                                # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:228]: This should be in host

**Comment by xhochy** [comment]
Should we put it instead into `host` to have the right `run_exports`? Otherwise this is confirmed by the log on master:

```
2021-05-17T16:07:18.3108669Z WARNING (clang-12,Library/bin/clang-12.exe): $RPATH/zlib.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.
2021-05-17T16:07:18.3110087Z .. is this binary repackaging?
```

**Comment by xhochy** [comment]
`clang`, `clangxx`, and`libclang` also need this according to the log.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -252,8 +254,11 @@ outputs:
     script: install_clang_symlinks.sh  # [unix]
     script: install_clang_symlinks.bat  # [win]
     requirements:
-      - {{ pin_subpackage("clang-" ~ major_version, exact=True) }}
-      - gcc_impl_{{ target_platform }}  # [linux]
+      host:
+        - zlib  # [win]
+      run:
+        - {{ pin_subpackage("clang-" ~ major_version, exact=True) }}
+        - gcc_impl_{{ target_platform }}  # [linux]
```
Comment (h-vetinari) [inline recipe/meta.yaml:261]: To be honest, I don't know what a build/host/run-less `requirements` defaults to and the [docs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#requirements-section) don't mention it either. So not sure whether
```
        - {{ pin_subpackage("clang-" ~ major_version, exact=True) }}
        - gcc_impl_{{ target_platform }}  # [linux]
```
should be host or run

Comment (isuruf) [reply recipe/meta.yaml:261]: Should be both.

Comment (isuruf) [reply recipe/meta.yaml:261]: It defaults to both host and run.

**Comment by h-vetinari** [comment]
@isuruf @xhochy 
This is green - hope I managed to do everything as asked/suggested.

**Pull request merged**